### ISSUE TITLE
missing lit tool from Fedora repository work around

### ISSUE THREAD

**Issue by baston007** [msoos/cryptominisat#287]
Currently there is no RPM package for lit within the Fedora 22 repository.  Without lit the build_norm.sh script will fail.  Is there somewhere else that a lit package is available?  As a temporary work-around the following may work.  Would you please add it to the build directory.

Rob

$ cat build_notest.sh
#!/bin/bash

set -e

rm -rf cm* CM* lib* cryptomini* Testing* tests* py-lib include
rm -rf tests
cmake -DENABLE_TESTING=OFF ..
make -j16
# make test
# cd ../tests/simp-checks/
# ./checks.py -l ../../build/cryptominisat4 testfiles/*
# cd ../../build/

**Comment by baston007** [comment]
The bottom 4 lines should be commented out.  For some reason the "#" at the start of these 4 lines did not cut and paste correctly.

Rob

**Comment by msoos** [comment]
I'm sorry, but this is trivial to fix. Install pip:

```
sudo yum install python-pip
```

And then

```
pip install lit
```

Also, please, please look at .travis.yml and travis-cmake.sh A lot of this is all there.

**Comment by baston007** [comment]
Hi Mate,
       Little by little I am getting there.  I have installed python-pip
    then was able to install lit.  Now when I run "build_norm.sh" I get
    the following error messages:
    
    $ ./build_norm.sh
    -- You can choose the type of build, options
    are:Debug;Release;RelWithDebInfo;MinSizeRel
    -- Doing a RelWithDebInfo build
    -- The C compiler identification is GNU 5.1.1
    -- The CXX compiler identification is GNU 5.1.1
    -- Check for working C compiler: /usr/lib64/ccache/cc
    -- Check for working C compiler: /usr/lib64/ccache/cc -- works
    -- Detecting C compiler ABI info
    -- Detecting C compiler ABI info - done
    -- Detecting C compile features
    -- Detecting C compile features - done
    -- Check for working CXX compiler: /usr/lib64/ccache/c++
    -- Check for working CXX compiler: /usr/lib64/ccache/c++ -- works
    -- Detecting CXX compiler ABI info
    -- Detecting CXX compiler ABI info - done
    -- Detecting CXX compile features
    -- Detecting CXX compile features - done
    -- Performing Test HAVE_FLAG_-Wall
    -- Performing Test HAVE_FLAG_-Wall - Success
    -- flag -Wall supported
    -- Performing Test HAVE_FLAG_-Wextra
    -- Performing Test HAVE_FLAG_-Wextra - Success
    -- flag -Wextra supported
    -- Performing Test HAVE_FLAG_-Wunused
    -- Performing Test HAVE_FLAG_-Wunused - Success
    -- flag -Wunused supported
    -- Performing Test HAVE_FLAG_-pedantic
    -- Performing Test HAVE_FLAG_-pedantic - Success
    -- flag -pedantic supported
    -- Performing Test HAVE_FLAG_-Wsign-compare
    -- Performing Test HAVE_FLAG_-Wsign-compare - Success
    -- flag -Wsign-compare supported
    -- Performing Test HAVE_FLAG_-Wtype-limits
    -- Performing Test HAVE_FLAG_-Wtype-limits - Success
    -- flag -Wtype-limits supported
    -- Performing Test HAVE_FLAG_-Wuninitialized
    -- Performing Test HAVE_FLAG_-Wuninitialized - Success
    -- flag -Wuninitialized supported
    -- Performing Test HAVE_FLAG_-Wno-deprecated
    -- Performing Test HAVE_FLAG_-Wno-deprecated - Success
    -- flag -Wno-deprecated supported
    -- Performing Test HAVE_FLAG_-Wstrict-aliasing
    -- Performing Test HAVE_FLAG_-Wstrict-aliasing - Success
    -- flag -Wstrict-aliasing supported
    -- Performing Test HAVE_FLAG_-Wpointer-arith
    -- Performing Test HAVE_FLAG_-Wpointer-arith - Success
    -- flag -Wpointer-arith supported
    -- Performing Test HAVE_FLAG_-Wheader-guard
    -- Performing Test HAVE_FLAG_-Wheader-guard - Failed
    -- flag -Wheader-guard NOT supported
    -- flag -Wpointer-arith supported
    -- Performing Test HAVE_FLAG_-Wformat-nonliteral
    -- Performing Test HAVE_FLAG_-Wformat-nonliteral - Success
    -- flag -Wformat-nonliteral supported
    -- Performing Test HAVE_FLAG_-Winit-self
    -- Performing Test HAVE_FLAG_-Winit-self - Success
    -- flag -Winit-self supported
    -- Performing Test HAVE_FLAG_-Wparentheses
    -- Performing Test HAVE_FLAG_-Wparentheses - Success
    -- flag -Wparentheses supported
    -- Performing Test HAVE_FLAG_-Wunreachable-code
    -- Performing Test HAVE_FLAG_-Wunreachable-code - Success
    -- flag -Wunreachable-code supported
    -- Performing Test HAVE_FLAG_-ggdb3
    -- Performing Test HAVE_FLAG_-ggdb3 - Success
    -- flag -ggdb3 supported
    -- GIT hash found: GIT-notfound
    -- PROJECT_VERSION: 4.5.3
    -- PROJECT_VERSION_MAJOR: 4
    -- PROJECT_VERSION_MINOR: 5
    -- PROJECT_VERSION_PATCH: 3
    -- Boost version: 1.57.0
    -- Found the following Boost libraries:
    --   unit_test_framework
    --   program_options
    -- Looking for include file pthread.h
    -- Looking for include file pthread.h - found
    -- Looking for pthread_create
    -- Looking for pthread_create - not found
    -- Looking for pthread_create in pthreads
    -- Looking for pthread_create in pthreads - not found
    -- Looking for pthread_create in pthread
    -- Looking for pthread_create in pthread - found
    -- Found Threads: TRUE  
    -- Not compiling detailed statistics. Leads to faster system
    -- Found ZLIB: /usr/lib64/libz.so (found version "1.2.8") 
    -- OK, Found ZLIB!
    -- Valgrind Prefix: 
    -- Found VALGRIND: /usr/include/valgrind  
    -- OK, Found Valgrind. Using valgrind client requests to mark freed
    clauses in pool as undefined
    -- Found PkgConfig: /usr/bin/pkg-config (found version "0.28") 
    -- Found m4ri: /usr/lib64/libm4ri.so  
    -- OK, Found M4RI lib at /usr/lib64/libm4ri.so and includes at
    /usr/include
    -- All defines at startup:  -DBOOST_TEST_DYN_LINK -DUSE_ZLIB
    -DUSE_VALGRIND -DUSE_M4RI
    -- Found PythonInterp: /usr/bin/python2.7 (found suitable version
    "2.7.10", minimum required is "2.7") 
    -- Found PythonLibs: /usr/lib64/libpython2.7.so (found suitable
    version "2.7.10", minimum required is "2.7") 
    -- OK, found python interpreter, libs and header files ->
    building python plugin
    -- Will run python test from
    /home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/build/py-lib
    directory
    -- Testing is enabled
    -- Found PythonInterp: /usr/bin/python2.7 (found version "2.7.10") 
    -- NOTE: if adding the 'gtest' subdirectory fails, you need to issue
    'git submodule init' and 'git submodule update'
    CMake Error at tests/CMakeLists.txt:47 (add_subdirectory):
      The source directory
    
       
/home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/utils/gtest
    
      does not contain a CMakeLists.txt file.
    
    
    CMake Error at tests/CMakeLists.txt:72 (add_subdirectory):
      The source directory
    
       
/home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/utils/cnf-utils
    
      does not contain a CMakeLists.txt file.
    
    
    CMake Error at tests/CMakeLists.txt:73 (add_subdirectory):
      The source directory
    
       
/home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/utils/sha1-sat
    
      does not contain a CMakeLists.txt file.
    
    
    -- Will run solver test from
    /home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/build/tests/swig
    directory
    -- Configuring incomplete, errors occurred!
    See also
"/home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/build/CMakeFiles/CMakeOutput.log".
    See also
"/home/baston/kryptonite/misc/cryptominisat-master_oct19_2015_fb506ee/build/CMakeFiles/CMakeError.log".
    [baston@desk build]$ 
    
    
    
    
    It appears that the needed git files are missing from the download. 
    Since these files all begin with a dot ".", did zip miss them when
    the directory was packaged up?  When I run "git submodile init"  git
    complains that there are no git files all the way up the directory
    structure.
    
    The directories /utils/gtest, /utils/sha1-sat, /utils/Outputcheck,
    /utils/licensecheck and /utils/cnf-utils are all empty.
    
    Rob
    
    
    On 19/10/15 02:39 PM, Mate Soos wrote:
    
    
      I'm sorry, but this is trivial to fix. Install pip:
      sudo yum install python-pip

      And then
      pip install lit

      Also, please, please look at .travis.yml and travis-cmake.sh A
        lot of this is all there.
      —
        Reply to this email directly or view
          it on GitHub.

**Comment by msoos** [comment]
Please read what that says above. It says, specifically:

```
 -- NOTE: if adding the 'gtest' subdirectory fails, you need to issue
    'git submodule init' and 'git submodule update'
```

**Comment by msoos** [comment]
Again, PLEASE READ THE TRAVIS.YML AND TRAVIS-CMAKE.SH. I cannot stress that enough. I will not reply anymore unless you read it IN FULL.

**Comment by a1880** [comment]
@baston007 : The "#" at beginning of a line is interpreted as headline by markdown. Enclose it in backticks as escape. http://stackoverflow.com/questions/20532546/escape-pound-or-number-sign-in-github-issue-tracker

**Comment by msoos** [comment]
Also, you can just quote stuff like that, using the backticks. I fixed it for you...

**Comment by msoos** [comment]
Ah, I see what's going on. You are replying to the mail. I suggest you don't do that :) You need to follow the link that's in the email and then update the issue from there. Using the email robs you of a lot of very good things, including proper markup. So I can't fix the markup for you.

**Comment by baston007** [comment]
After much head scratching and reading I have found the Travis-ci.com web site.  It looks like I will be doing some more reading.
   It appears that the "git submodule init" and "git submodule update" commands first require access to the Travis-ci web site.  Without this "git submodule init" generates the following error message:

$ git submodule init
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
$

Even though there is a .gitmodules file in the current directory.

Rob

**Comment by msoos** [comment]
No, it needs no such thing. Just issue the commands in the right directory, the directory where you git clone-d cryptominisat.

**Comment by baston007** [comment]
I think I found the root of my problem.  I have been downloading the zip file from github.  This zip file does not include the .git directory needed for the "git submodule" commands to work.  I just downloaded cryptominisat again, but this time using "git clone".  This download did include the .git directory.  Now I can run "git submodule init" and "git submodule update".  Now the empty directories are filled in.  I was still missing "swig", which I downloaded and installed.  The "build_norm.sh" script is looking for "swig3.0", but the latest swig, (version 3.0.7), is still called just "swig".  I installed a soft link so "swig3.0" pointed to "swig" in the /usr/local/bin directory.  That got me a lot farther, but not all the way.  I am still missing a "testfiles" directory.  After passing 7 tests I get the folowing error message:

Executing: ../../build/cryptominisat4 --zero-exit-status -p1 testfiles/* simp.out
ERROR! Could not open file 'testfiles/*' for reading: No such file or directory

Where should the "testfiles" directory be located?

In conclusion, please add the ".git" directory into the zip file.  Have the "build_norm.sh" script look for either "swig3.0" or "swig".  Add the "testfiles" directory to both the zip file and the git structure.  Thank.

Rob

**Comment by msoos** [comment]
You should never use the ZIP file. I have said that before multiple times. It's bad because you don't know the exact revision, it's bad because you don't know what you have changed, and it's bad because you can't do testing. As a rule, if you are not using the END PRODUCT, i.e. the released version, you MUST git clone.

Also, the ZIP file is NOT under my control so I CANNOT add anything to it -- and it would be *REALLY* bad practice to add the .git directory in there -- that can be gigabytes of stuff

as for that test -- you didn't see in the travis-cmake.sh that it clones a repository before executing it. The repository is called `testfiles`. Again, please read those files carefully. Thanks.

**Comment by baston007** [comment]
SOLVED.   I now am using the git version and have installed the various needed software packages.  When I run "build_norm.sh", the final line shows "ALL PASSED".

Rob
